packages/teams-js/src/internal/communication.ts

```diff
@@ -63,7 +63,7 @@ export function initializeCommunication(validMessageOrigins: string[] | undefine
 
   // If we are in an iframe, our parent window is the one hosting us (i.e., window.parent); otherwise,
   // it's the window that opened us (i.e., window.opener)
-  Communication.currentWindow = Communication.currentWindow || window;
```

Comment:
While not exact, I think it's possible to add a git hook that runs against the code being checked in that looks for the word "window" and blurts out a warning message before you push that code up. As a potential option for this,

Reply:
I think you want to use eslint rather than a githook for this because I'm pretty sure that git hooks are local and so everyone who ever contributed would have to know/remember to apply it to their local environment.

Eslint was made for this sort of thing and will automatically apply to everyone.

[https://stackoverflow.com/questions/40977713/how-to-forbid-a-specific-named-function-with-eslint](https://stackoverflow.com/questions/40977713/how-to-forbid-a-specific-named-function-with-eslint)

or

[https://www.npmjs.com/package/eslint-plugin-ban](https://www.npmjs.com/package/eslint-plugin-ban)

Reply:
(note: I think it would be okay to come back an add the eslint rule in a subsequent change to this; doesn't necessarily need to be included in this change)